### ISSUE TITLE
replay recording not user friendly

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#919]
i am struggling still with the replay recording function. partly this is because i don't often remember to start a recording before a performance, or if i remember it's too late - it would be too stressful for the other players to have the stage cleared at this point (they may have already prepared drawings, texts, etc).
so my work around has been to archive the performance after the show, then at least there is a recording that ends at that time, so i should be able to go to the last part of the recording and find the show.
BUT the scrub bar is pretty unusable. i cannot click precisely somewhere. it seems like i click near the end & i'm still days away from the performance - even setting it to run at x32, i'm waiting and waiting and the performance is never coming. so i try to click closer to the end - but it takes it as a click at the very end and goes back to the start. it's just not possible to click within the last few days of the recording & also not realistic to sit and watch nothing, waiting and waiting for the performance to start.
i tried using the forward and back arrows but still did not find a way to get to the start of the performance.

i'm aware that this is partly my own fault for not remembering to start a recording at a suitable time, but it's quite frustrating to know that these recordings are there but not really accessible or useable. i am not sure what the solution might be, but here are some thoughts:

- a much wider scrub bar: if the scrub bar extended right across the bottom of the stage (like it does with a video) it might be easier to click closer to the end.
- date markers in the log: could UpStage insert a marker at midnight each day & there be a way to jump from marker to marker?
- a way to cut the log file into smaller pieces, maybe also using this kind of marker - e.g. to be able to save a section of the log from specific dates or lines, as a separate reply recording
- a way to start the recording without having to clear the stage ...

**Comment by vickis** [comment]
Wider scrub bar and markets would be useful
Date stamp could help but which Timezone midnight?
Building clear stage into the re show routine would be good for ensuring
the stages are ready and helping with the post recording archival process

**Comment by helenvarley** [comment]
i don't think it matters which timezone, it would just need to make it easier to get close to what you want, or to at least know approximately how far away you might be. at the moment there are some numbers that don't mean anything to the user, it's not possible to know whether you are near to what you want or not.

looking at the archive from yesterday, it says that the length is "1 day, 3 hours, 32 minutes, 28 seconds" - which is interesting as i thought the archive went from the time of the last recording or archiving. nothing has been captured during the month of april. i presume there was some activity on this stage during that time, as others were uploading media & testing streams. so how does the archive function decide where to start from?

**Comment by vickis** [comment]
yes agree

maybe GMT - to acknowledge the origin of timezones generally - in the
digital timeless space of online :)

it would be good to not automatically archive weeks of empty stage use - to
keep the servers clear :)

**Comment by helenvarley** [comment]
>it would be good to not automatically archive weeks of empty stage use - to keep the servers clear :)

yes - but how could that work? how would it distinguish between an empty stage without anything happening, and a dramatic pause in a performance? i guess if there are players present on the stage that is a signal that something is happening, but at the moment the recording doesn't include player/audience count; also i sometimes leave a stage open in a browser tab while doing other things, so i'm on the stage but not actually doing anything.

i don't think the size of the log file would be much reduced by leaving out the bits where nothing is happening, since there would be no information to record. the main thing at the moment is to improve how we can navigate through to the bits we want.

**Comment by helenvarley** [comment]
discussing this with vicki after the 20th birthday parties, & again having difficulty getting to the bits of the recording we want.
wondering if instead of having a "start recording" function for when a performance is about begin, could it be an "insert marker" function, so that the stage doesn't have to be cleared but we can later find the point in the recording. 
the "archive performance" function would work as it does now, ending a recording and sweeping the stage.
we would still need a way to make simple edits to a replay recording, so that we could save out the relevant part as a separate recording.

**Comment by helenvarley** [comment]
closing this as i have created #1447 to develop requirements for a rewrite of the replay recording feature.
